lib/spack/spack/test/buildrequest.py

```diff
@@ -23,7 +23,7 @@ def test_build_request_errors(install_mockery):
     with pytest.raises(ValueError, match='must be a package'):
         inst.BuildRequest('abc', {})
 
-    pkg = spack.repo.get('trivial-install-test-package')
+    pkg = spack.repo.path.get('trivial-install-test-package')
```

Comment:
This could just be converted to ensure that `get()` raises with an abstract `Spec`, or with a non-`Spec` object, since auto-spec'd things will definitely be abstract.  That would avoid the overhead of automatic `Spec` instantiation in `get()`, too.